src/App.tsx

```diff
@@ -15,36 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+const handleSort = (
+  type: SortType,
+  setGoods: React.Dispatch<React.SetStateAction<string[]>>,
```

Comment:
- The utility function shouldn't know anything about the state, just gets goods, sorts and returns them
- utils should be a pure functions https://www.geeksforgeeks.org/pure-functions-in-javascript/

Reply:
Not fixed!